domino_data/_feature_store/git.py

```diff
@@ -0,0 +1,72 @@
+"""Containing APIs to interact with feast git repository"""
+
+from git import FetchInfo, PushInfo, Repo
+
+from ..logging import logger
+from .exceptions import GitPullError, GitPushError
+
+_INVALID_PULL_FLAGS = [FetchInfo.ERROR, FetchInfo.REJECTED]
+_INVALID_PUSH_FLAGS = [
+    PushInfo.ERROR,
+    PushInfo.REJECTED,
+    PushInfo.REMOTE_FAILURE,
+    PushInfo.REMOTE_REJECTED,
+]
+
+
+def pull_repo(repo: Repo) -> None:
+    """Pull feast repo.
+
+    Args:
+        repo: The feast Git Repo object.
+
+    Raises:
+        GitPullError: if fails to pull the repo
+    """
+    logger.info(f"Commit Sha before pulling:{repo.head.object.hexsha}")
+    logger.info("Pulling the repo......")
+    pull_result = repo.remotes.origin.pull()
+    result_flag = pull_result[0].flags
+    if result_flag in _INVALID_PULL_FLAGS:
+        raise GitPullError(f"Failed to pull the repo with error flag {result_flag}")
+    logger.info("Finished pulling the repo.")
+    logger.info(f"Commit Sha after pulling:{repo.head.object.hexsha}")
+
+
+def push_to_git(repo: Repo) -> None:
```

Comment:
shouldn't this return the new commit?

Reply:
Don't bother to return it cause the new commit id is updated in the repo object.